_delphi_utils_python/delphi_utils/validator/datafetcher.py

```diff
@@ -159,9 +163,8 @@ def fetch_api_reference(data_source, start_date, end_date, geo_type, signal_type
     Formatting is changed to match that of source data CSVs.
     """
     with warnings.catch_warnings():
-        warnings.simplefilter("ignore")
-        api_df = covidcast.signal(
-            data_source, signal_type, start_date, end_date, geo_type)
+        api_df = covidcast_wrapper.signal(data_source, signal_type, start_date, end_date, geo_type)
```

Comment:
I dont think this context manager will still ignore warnings if you remove the `.simplefilter()` call.  I also suspect that the `delphi-epidata` lib will not produce a warning where the `covidcast` lib did, so you can probably remove the c.m. entirely.  Do you know what the text of the warning from `covidcast.signal()` actually is?  Is it something we need to worry about?

Reply:
the warnings were either no data or too much data:
in def _fetch_epidata():
```
        # Two possible error conditions: no data or too much data.
        if day_data["message"] == "no results":
            warnings.warn(f"No {data_source} {signal} data found on {day_str} "
                          f"for geography '{geo_type}'",
                          NoDataWarning)
        if day_data["message"] not in {"success", "no results"}:
            warnings.warn(f"Problem obtaining {data_source} {signal} data on {day_str} "
                          f"for geography '{geo_type}': {day_data['message']}",
                          RuntimeWarning)
```                          

 if the response for that date was too big, I recall Dmitry mention how back in the day the limit used to be 5k per response and that was the main reason behind the covidcast api having a for loop for each date and truncating the rest or something

Reply:
The current limit is 10M rows returned per request, we shouldnt have problems with it, but it is still worth catching the error condition.